visualization/dashboard/src/MLIDashboard/ExplanationDashboard.tsx

```diff
@@ -200,6 +214,128 @@ export class ExplanationDashboard extends React.Component<IExplanationDashboardP
         };
     }
 
+    private static validateInputs(props: IExplanationDashboardProps, modelMetadata: IExplanationModelMetadata): string | undefined {
+        const classLength = modelMetadata.classNames.length;
+        const featureLength = modelMetadata.featureNames.length;
+        let rowLength: number;
+        if (props.trueY) {
+            rowLength = props.trueY.length;
+        }
+        if (props.predictedY) {
+            const length = props.predictedY.length;
+            if (rowLength === undefined) {
+                rowLength = length;
+            }
+            if (length !== rowLength) {
+                return `Inconsistent dimensions. Predicted y has dimension [${length}], expected [${rowLength}]`;
+            }
+        }
+        if (props.probabilityY) {
+            if (!Array.isArray(props.probabilityY)) {
+                return `Predicted probability not an array. Expected array of dimension [${rowLength} x ${classLength}]`;
+            }
+            const length = props.probabilityY.length;
+            if (rowLength === undefined) {
+                rowLength = length;
+            }
+            if (length !== rowLength) {
+                return `Inconsistent dimensions. Predicted y has length [${length}], expected [${rowLength}]`;
+            }
+            if (length === 0) {
+                return "Predicted probability input not a non-empty array";
+            }
+            const cLength = props.probabilityY[0].length;
+            if (cLength !== classLength) {
+                return `Inconsistent dimensions. Predicted probability has dimensions [${length} x ${cLength}], expected [${rowLength} x ${classLength}]`;
+            }
+            if (!props.probabilityY.every(row => row.length === classLength)) {
+                return `Inconsistent dimensions. Predicted probability has rows of varying length`;
+            }
+        }
+        if (props.testData) {
+            if (!Array.isArray(props.testData)) {
+                return `Eval dataset not an array. Expected array of dimension [${rowLength} x ${featureLength}]`;
+            }
+            const length = props.testData.length;
+            if (rowLength === undefined) {
+                rowLength = length;
+            }
+            if (length !== rowLength) {
+                return `Inconsistent dimensions. Eval dataset has length [${length}], expected [${rowLength}]`;
+            }
+            if (length === 0) {
+                return "Eval dataset not a non-empty array";
+            }
+            const fLength = props.testData[0].length;
+            if (fLength !== featureLength) {
```

Comment:
I think some of this code that checks values could be refactored out into some small utilities that do the check and take in the name of the thing being checked, as I see very similar if statements checking the same stuff several times

Reply:
definitely, but the old dashboard is in maintenance-mode, so this is more of a bug fix than adding a new feature.